Library/ViewModels/RewardCardContainerViewModel.swift

```diff
@@ -32,16 +34,31 @@ public final class RewardCardContainerViewModel: RewardCardContainerViewModelTyp
 
     let reward: Signal<Reward, Never> = projectAndRewardOrBacking
       .map { project, rewardOrBacking -> Reward in
-        rewardOrBacking.left
-          ?? rewardOrBacking.right?.reward
-          ?? backingReward(fromProject: project)
-          ?? Reward.noReward
+        switch rewardOrBacking {
+        case let .left(reward):
+          return reward
+        case let .right(backing):
+          return Library.reward(from: backing, inProject: project)
+        }
       }
 
     let projectAndReward = Signal.zip(project, reward)
 
     self.currentRewardProperty <~ reward
 
+    self.configureNoRewardGradientView = reward
+      .filter { _ in featureGoRewardlessIsEnabled() }
+      .map { $0.isNoReward }
```

Comment:
So this will only emit if the feature flag is _enabled_, in which case the gradient view wouldn't be removed if the feature were _disabled_ perhaps by backgrounding/foregrounding on the carousel?

Probably an edge case but would it be solved by removing the `filter` and the `map` being `.map { $0.isNoReward &&  featureGoRewardlessIsEnabled() }`?

Reply:
Yeah good point!